File: common/src/main/java/com/ssblur/scriptor/item/Ingredient.java
package com.ssblur.scriptor.item;

import net.minecraft.ChatFormatting;
import net.minecraft.network.chat.Component;
import net.minecraft.world.item.Item;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.item.TooltipFlag;

import java.util.List;

public class Ingredient extends Item {
  String[] description;
  public Ingredient(Properties properties, String... description) {
    super(properties);
    this.description = description;
  }

  @Override
  public void appendHoverText(ItemStack itemStack, TooltipContext level, List<Component> list, TooltipFlag tooltipFlag) {
    super.appendHoverText(itemStack, level, list, tooltipFlag);

    for(var string: description)
      list.add(Component.translatable(string).withStyle(ChatFormatting.GRAY));
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/item/Ingredient.kt
package com.ssblur.scriptor.item

import net.minecraft.ChatFormatting
import net.minecraft.network.chat.Component
import net.minecraft.world.item.Item
import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.TooltipFlag

class Ingredient(properties: Properties, vararg val description: String): Item(properties) {
  override fun appendHoverText(
    itemStack: ItemStack,
    level: TooltipContext,
    list: MutableList<Component>,
    tooltipFlag: TooltipFlag
  ) {
    super.appendHoverText(itemStack, level, list, tooltipFlag)

    for (string in description) list.add(Component.translatable(string).withStyle(ChatFormatting.GRAY))
  }
}
